integration-test/src/test/java/com/datastrato/gravitino/integration/test/container/BaseContainer.java

```diff
@@ -124,7 +125,33 @@ public void start() {
     container.start();
   }
 
-  protected abstract boolean checkContainerStatus(int retryLimit);
+  protected abstract boolean checkContainerStatus(int timeoutMillis);
+
+  protected boolean checkContainerStatusWithRetry(int timeoutMillis, BooleanSupplier... checker) {
+    int nRetry = 0;
+    int sleepTimeMillis = 5_000;
+    int totalSleepTimeMillis = 0;
+
+    // retry until timeout
+    for (BooleanSupplier check : checker) {
+      while (totalSleepTimeMillis < timeoutMillis) {
+        if (check.getAsBoolean()) {
+          break;
+        }
+
+        sleepTimeMillis = sleepTimeMillis * (int) Math.pow(2, nRetry++);
```

Comment:
1. Sleep time is too long, This computation method is increasing too rapidly. 5 , 10, 20, 40 ... 
2. Use shift operator instead of power

Reply:
> 1. Sleep time is too long, This computation method is increasing too rapidly. 5 , 10, 20, 40 ...

How about 1, 2, 4, 8, 16 ... ? What are the issues with rapid growth? What is your opinion?

> 2. Use shift operator instead of power

How does it differ from the current implementation?